### ISSUE TITLE
Grafana - Rename by Regex doesnt work with global patterns

### ISSUE THREAD

**Issue by HenriqueLBorges** [grafana/grafana#46894]
Hi, I have the following field in my query:

Example: `{method="GET", status="200", uri="/v2/path1/jhkdsahkjdhas89d2182137/path2/3213213dsagg4312321/path3"}`

I want to apply a transformation in Grafana to remove the path parameters from URI so I could properly group them.

But when I apply the `Rename By Regex` transformation, it applies to the first match. I am unable to use the `\g` to set the global flag in the Regex.

My Regex expression: `/(\/([A-Za-z]*[0-9]+)+\/)/g`
My Replace value: `path-param`
The result I expected: `{method="GET", status="200", uri="/v2/path1/path-param/path2/path-param/path3"}`
The result I receive: `{method="GET", status="200", uri="/v2/path1/path-param/path2/3213213dsagg4312321/path3"}`

The only workaround I found is to repeat the transformation which is not feasible since I have multiple endpoints with different numbers of path-params each one.

**Comment by zuchka** [comment]
Thanks for creating this issue, @HenriqueLBorges

I don't know if the current `Rename by Regex` transformation can access the [global prop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global) on RegExp objects.

The `g flag` is mentioned in the `stringToJsRegex` function:

https://github.com/grafana/grafana/blob/yaelle%2Fa11y%2Ffix-CollapsibleSection/packages/grafana-data/src/text/string.ts#L33-L34

**Comment by ashharrison90** [comment]
ok so bit of a brain dump...

originally i thought this just wasn't supported and was a feature request, however the more i look at the `renameByRegex` transformation i'm starting to think it's a bug we should fix. some reasons:
- whilst the transformation itself just [creates a new RegExp object](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/transformations/transformers/renameByRegex.ts#L48), the frontend for the transformation determines [whether a regex is valid using a different function](https://github.com/grafana/grafana/blob/main/public/app/features/transformers/editors/RenameByRegexTransformer.tsx#L40)
- this leads to some bugs/inconsistencies:
  - there is no way to match on `/foo` for example, because if you type `/foo` the frontend throws an error saying it's an invalid pattern because the `stringToJsRegex` function is expecting you to complete the regex (e.g. write something like `/\/foo/`), but the transformation doesn't accept that syntax...
  - it's different to [what we do in value mappings](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/valueMappings.ts#L59). there we use the `stringToJsRegex` function.

my proposal would be to make a change to use the `stringToJsRegex` function in the `renameByRegex` transformation. that way, people would be able to add more complicated regex expressions like `/foo/g` . the downside to this is it has the potential to break some existing transformations, because whilst the `stringToJsRegex` function accepts a non-regex string, it will convert it to `new RegExp(^<expression>$)` instead of `new RegExp(<expression>)` which is slightly different.

there's a branch here with the change: https://github.com/grafana/grafana/compare/ash/46894?expand=1

tl;dr: i think we should fix this. the fix is technically a breaking change, so we'll have to wait until grafana 9 to deliver it.
